packages/x-tree-view/src/internals/useTreeView/useTreeView.ts

```diff
@@ -65,39 +71,46 @@ export const useTreeView = <
   const innerRootRef: React.RefObject<HTMLUListElement> = React.useRef(null);
   const handleRootRef = useForkRef(innerRootRef, rootRef);
 
-  const contextValue = useTreeViewBuildContext({
-    plugins,
-    instance,
-    publicAPI,
-    rootRef: innerRootRef,
-  });
+  const storeRef = React.useRef<TreeViewStore<TSignaturesWithCorePluginSignatures> | null>(null);
+  if (storeRef.current == null) {
+    globalId += 1;
```

Comment:
I'm a bit surprised you don't get the same lint error as me on charts for this line

> Unexpected reassignment of a variable which was defined outside of the component. Components and hooks should be pure and side-effect free, but variable reassignment is a form of side-effect. If this variable is used in rendering, use useState instead. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#side-effects-must-run-outside-of-render)eslint(react-compiler/react-compiler)

Reply:
Because only the charts have those linting rules enabled right now

Reply:
The core has the same "problem" with `useId`: https://github.com/mui/material-ui/blob/e84fc98f088c8b3e2661384b27f240ce5cf0a0af/packages/mui-utils/src/useId/useId.ts#L40C1-L41C1